lib/benchee/formatters/json.ex

```diff
@@ -55,12 +56,25 @@ defmodule Benchee.Formatters.JSON do
       ...>            mode:          400,
       ...>            sample_size:   8,
       ...>            percentiles:   %{99 => 900}
+      ...>          },
+      ...>          memory_usage_statistics: %Benchee.Statistics{
+      ...>            average:       500.0,
+      ...>            ips:           nil,
+      ...>            std_dev:       200.0,
+      ...>            std_dev_ratio: 0.4,
+      ...>            std_dev_ips:   nil,
+      ...>            median:        450.0,
+      ...>            minimum:       200,
+      ...>            maximum:       900,
+      ...>            mode:          nil,
+      ...>            sample_size:   8,
+      ...>            percentiles:   %{99 => 900}
       ...>          }
       ...>        },
       ...>      ]
       ...>    }
       iex> Benchee.Formatters.JSON.format(suite, %{file: "my_file.json"})
-      %{"Some Input" => "{\\"run_times\\":{\\"My Job\\":[200,400,400,400,500,500,700,900]},\\"sort_order\\":[\\"My Job\\"],\\"statistics\\":{\\"My Job\\":{\\"average\\":500.0,\\"ips\\":2.0e3,\\"maximum\\":900,\\"median\\":450.0,\\"minimum\\":200,\\"mode\\":400,\\"percentiles\\":{\\"99\\":900},\\"sample_size\\":8,\\"std_dev\\":200.0,\\"std_dev_ips\\":800.0,\\"std_dev_ratio\\":0.4}}}"}
+      %{"Some Input" => "{\\"memory_usage_statistics\\":{\\"My Job\\":{\\"average\\":500.0,\\"ips\\":null,\\"maximum\\":900,\\"median\\":450.0,\\"minimum\\":200,\\"mode\\":null,\\"percentiles\\":{\\"99\\":900},\\"sample_size\\":8,\\"std_dev\\":200.0,\\"std_dev_ips\\":null,\\"std_dev_ratio\\":0.4}},\\"memory_usages\\":{\\"My Job\\":[200,400,400,400,500,500,700,900]},\\"run_time_statistics\\":{\\"My Job\\":{\\"average\\":500.0,\\"ips\\":2.0e3,\\"maximum\\":900,\\"median\\":450.0,\\"minimum\\":200,\\"mode\\":400,\\"percentiles\\":{\\"99\\":900},\\"sample_size\\":8,\\"std_dev\\":200.0,\\"std_dev_ips\\":800.0,\\"std_dev_ratio\\":0.4}},\\"run_times\\":{\\"My Job\\":[200,400,400,400,500,500,700,900]},\\"sort_order\\":[\\"My Job\\"]}"}
```

Comment:
As we're talking breaking changes - I think we should consider changing the whole JSON formatter layout for 1.0. It'd make more/most sense if it mirrored `scenarios` from the main `suite` more closely so that we don't have to learn another data structure. What do you think?

Reply:
@PragTob so the format would be something like this?

```elixir
[
  %{
    "name" => "Sleep",
    "input_name" => "first_input",
    "memory_usages" => [],
    "memory_usage_statistics" => %{},
    "run_times" => [],
    "run_time_statistics" => %{}
  },
  %{
    "name" => "Sleep",
    "input_name" => "second_input",
    "memory_usages" => [],
    "memory_usage_statistics" => %{},
    "run_times" => [],
    "run_time_statistics" => %{}
  },
  %{
    "name" => "List",
    "input_name" => "first_input",
    "memory_usages" => [],
    "memory_usage_statistics" => %{},
    "run_times" => [],
    "run_time_statistics" => %{}
  },
  %{
    "name" => "List",
    "input_name" => "second_input",
    "memory_usages" => [],
    "memory_usage_statistics" => %{},
    "run_times" => [],
    "run_time_statistics" => %{}
  }
]
```

I don't really know how folks use this formatter other than how we use it in the HTML formatter, so it's kind of hard to know which format would be better for users. For our use, though, I think this is totally fine, and indeed probably better.

Reply:
Hm yeah I'm thinking on it... some way to easily access by name might be convenient but in the end it's an easy select or find so I guess it should be fine